File: src/main/java/org/dxworks/githubminer/dto/response/repository/Repository.java
package org.dxworks.githubminer.dto.response.repository;

import com.google.api.client.json.GenericJson;
import com.google.api.client.util.Key;
import lombok.Data;
import org.dxworks.githubminer.dto.commons.User;

@Data
public class Repository extends GenericJson {
    @Key
    private Long id;
    @Key
    private String name;
    @Key("full_name")
    private String fullName;
    @Key
    private User owner;
    @Key("created_at")
    private String createdAt;
    @Key("updated_at")
    private String updatedAt;
    @Key
    private String language;
    @Key
    private String description;
}

File: github-miner-core/src/main/kotlin/org/dxworks/githubminer/dto/response/repository/Repository.kt
package org.dxworks.githubminer.dto.response.repository

import com.google.api.client.json.GenericJson
import com.google.api.client.util.Key
import org.dxworks.githubminer.dto.commons.User

class Repository : GenericJson() {
    @Key
    var id: Long? = null

    @Key
    var name: String? = null

    @Key("full_name")
    var fullName: String? = null

    @Key
    var owner: User? = null

    @Key("created_at")
    var createdAt: String? = null

    @Key("updated_at")
    var updatedAt: String? = null

    @Key
    var language: String? = null

    @Key
    var description: String? = null
}
